Fix issues with location restriction org policy and with subnet names

idofl opened the pull request:
1. Fix location restriction org policy by changing bucket exist test from POST to GET.
https://github.com/GoogleCloudPlatform/kubernetes-engine-windows-tools/issues/15
2. Add parameter workspace-bucket-location to select desired location for the bucket (default is "" which is US).
3. Fix issue when subnet is named differently than network by removing checking FW rule for subnet name. 
https://github.com/GoogleCloudPlatform/kubernetes-engine-windows-tools/issues/14

ibabou commented on gke-windows-builder/builder/builder/network.go:
```diff
@@ -70,21 +70,13 @@ func usingSharedVPC(netConfig *InstanceNetworkConfig) bool {
-	var networkUrl, subnetUrl string
-	subnetUrl = computeUrlPrefix + *netConfig.SubnetProject + "/global/networks/" + *netConfig.Subnet
-
-	if usingSharedVPC(netConfig) {
```
Thanks Ido for the PR! I don't have a full context about the change, but isn't this breaking the dependent logic in CheckProjectFirewalls? removing the specific handling of "usingSharedVPC()" when only SubnetProject is set & where *netConfig.NetworkProject == "". in that case the subnetUrl was the one returned. but now, we will generate a networkUrl from the empty NetworkProject, right?

idofl replied:
Apologies for the delay in responding. The original code checked FW rules in both the network and subnetwork projects. However, this code is off, as the network and subnetwork in Google Cloud need to be in the same project. There are only two options to deploy the VM:
- No Shared VPC: Network (+subnetwork), FW rule, and VM are all in the project. No need to specify network/subnetwork project, and if specified, they must be identical to the project ID.
- Shared VPC: VM uses project ID, while Network+Subnetwork+FW rule are in the host project (network project id). There is no need to also specify a subnetwork project.

Therefore looking for FW rules should only be done in a single project, where the network is declared. My code looks for a provided network project ID (shared vpc scenario), and if not provided, initializes it with the project's ID (no svpc).

I also went ahead and made the subnetwork-project obsolete, as we only need the network-project ID if we are using a shared vpc. I added checks in case both network and subnetwork projects were specified and are not identical.

One last thing, I noticed that looking for an existing instance only checks project and zone, ignoring the network configuration. I added a network+subnetwork filter to fix this.

idofl replied:
I also went ahead and made the subnetwork-project obsolete, as we only need the network-project ID if we are using a shared vpc. I added checks in case both network and subnetwork projects were specified and are not identical.

One last thing, I noticed that looking for an existing instance only checks project and zone, ignoring the network configuration. I added a network+subnetwork filter to fix this.

ibabou reviewed (changes requested):
Thanks for the PR, and apology for the delay to review! The changes to pass the workspaceBucketLocation through looks good to me.

idofl commented:
I also went ahead and made the subnetwork-project obsolete, as we only need the network-project ID if we are using a shared vpc. I added checks in case both network and subnetwork projects were specified and are not identical.

One last thing, I noticed that looking for an existing instance only checks project and zone, ignoring the network configuration. I added a network+subnetwork filter to fix this.

ibabou commented:
@idofl Thanks for the updates! It makes sense now, thanks for cleaning up/deprecating the usage then of subnetproject and checking that if both are specified, it should be the same (Also, if it's the one being passed, it'll be used). LGTM, I'll be merging the change.
